File: Algorithms/452. Minimum Number of Arrows to Burst Balloons/swift/solution.swift
class Solution {
    func findMinArrowShots(_ points: [[Int]]) -> Int {
        if points.isEmpty { return 0 }
        let points = points.sorted { $0[1] < $1[1] }

        var arrows = 1
        var firstEnd = points[0][1]
        for i in 1..<points.count {
            if points[i][0] > firstEnd {
                arrows += 1
                firstEnd = points[i][1]
            }
        }
        return arrows
    }
}

File: Algorithms/452. Minimum Number of Arrows to Burst Balloons/php/solution.php
class Solution
{

    /**
     * @param Integer[][] $points
     * @return Integer
     */
    function findMinArrowShots($points)
    {
        if (!count($points)) return 0;

        usort($points, function ($a, $b) {
            return $a[1] <=> $b[1];
        });

        $arrows = 1;
        $firstEnd = $points[0][1];
        for ($i = 1; $i < count($points); $i++) {
            if ($points[$i][0] > $firstEnd) {
                $arrows++;
                $firstEnd = $points[$i][1];
            }
        }
        return $arrows;
    }
}
